### DISCUSSION TITLE
How to send a webapp message to user

### DISCUSSION THREAD

**Discussion by guotie** [telegraf/telegraf#1903] [General]
Hi, I just want to send a webapp button to user, how to do it?

**Comment by guotie** [comment]
Now, I can send text + webapp to user, like this:

```
  bot.command('webapp', async (ctx) => {
    ctx.reply('webapp', {
        reply_markup: {
            inline_keyboard: [
                [
                {
                    text: 'web app test',
                    web_app: {
                        url: 'https://xxxxxx'
                    }
                }
            ]
            ]
        }
    })
  })
```

if I set the first arg of reply to '', error raised.

**Reply by MKRhere** [reply]
The text cannot be empty.

**Comment by MKRhere** [comment]
See the examples and read the docs here:
https://github.com/feathers-studio/telegraf-docs/tree/673ad547190568183ce699e42dd3920c4804cbec/examples/mini-apps
